### PR TITLE
Add Form Liquidity 

### PR DESCRIPTION
**Pull request by akintewe**
Fixed major issues and created new branch 

Images:

Screenshot 2024-11-30 at 10 15 37 AM

Screenshot 2024-11-30 at 10 15 49 AM

Screenshot 2024-11-30 at 10 15 56 AM

### CODE REVIEW

File: apps/mobile/src/components/search/TokenCard/index.tsx [hunk 1]
```diff
@@ -118,9 +116,17 @@ export const TokenCard: React.FC<LaunchCoinProps> = ({
             View token page
           </Button>
 
+          <Button onPress={() => {
```
Comment (MSghais) [inline apps/mobile/src/components/search/TokenCard/index.tsx:119]: Add check if you are the owner to see this button.

File: apps/mobile/src/hooks/launchpad/useAddLiquidity.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    coinAddress: string,
+    amount: string,
+    startingPrice: string,
+    lockTime: string
+  ) => {
+    try {
+      const deployCall = {
+        contractAddress: LAUNCHPAD_ADDRESS[constants.StarknetChainId.SN_SEPOLIA],
+        entrypoint: 'add_liquidity_unrug',
+        calldata: CallData.compile({
```
Comment (MSghais) [inline apps/mobile/src/hooks/launchpad/useAddLiquidity.ts:16]: Not the correct input, params etc of the launchpad.cairo, but thank you let me know on it later

**Review by MSghais — CHANGES_REQUESTED** [review]

File: apps/mobile/src/components/search/TokenCard/index.tsx (hunk 1 above)
Comment (akintewe) [reply apps/mobile/src/components/search/TokenCard/index.tsx:119]: Alright no problem

File: apps/mobile/src/components/AddLiquidityForm/validation.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import * as Yup from 'yup';
```
Comment (MSghais) [inline apps/mobile/src/components/AddLiquidityForm/validation.ts:1]: What's it's this lib?

**Review by MSghais — CHANGES_REQUESTED** [review]
The hooks is not ok, the form is not refacto, styling not done with theme, and also the YUP lib doesn't exist

**Comment by MSghais** [comment]
You need to take much more care of everything please. I've done so many review of this PR since the ODHack

**Comment by akintewe** [comment]
> You need to take much more care of everything please. I've done so many review of this PR since the ODHack

sorry for the continuous errors, please review my changes i have removed the yup and used the appropriate theme

**Review by MSghais — APPROVED** [review]
LFG good game! Even if not corresponding hooks/inputs for the contract and some refacto to do later ;)

Gg

**Pull request merged**